auth/utils/read-external-sources.ts

```diff
@@ -97,8 +96,31 @@ function filterPropertiesByServiceName(envObj: any, serviceName: string): any {
  * Pulls credentials from VCAP_SERVICES env property that IBM Cloud sets
  *
  */
-function getCredentialsFromCloud(serviceName: string): any {
-  const credentials = vcapServices.getCredentials(serviceName);
+function getVCAPCredentialsForService(name) {
+  if (process.env.VCAP_SERVICES) {
+    const services = JSON.parse(process.env.VCAP_SERVICES);
```

Comment:
I think it would be safer to wrap the JSON parse in a `try/catch` since it throws an error if anything about the JSON is malformatted. Not that it should be, since that env variable is generated by the cloud. So this is an optional change, up to you.

```suggestion
    let services;
    try {
      services = JSON.parse(process.env.VCAP_SERVICES);
    } catch () {
      return {};
    }
```

Reply:
Wrapping the JSON parse in a `try/catch` might eat up/silence the error, and the user wouldn't know their `VCAP_SERVICES` env is not parsed. So I think it's fine to let it fail if for some reason JSON is malformatted.

Reply:
Okay, that's a fair point. I just remembered that the core has it's own logger now. So another thing to consider would be printing an error log for a malformatted VCAP variable so the user knows about it. My only concern here is the code crashing even though the user typically cannot control the contents of `VCAP_SERVICES`.